### PR TITLE
Zip drag (needs review)

### PR DESCRIPTION
**Pull request by elisa-lj11**
This PR enables users to drag any zip folder into Interface, where it will be automatically unzipped and added to the Asset Browser. If the zip folder contains a .obj or .fbx, the model will be auto-added to world. The back button in Marketplaces has also been fixed and some of the styling for the marketplaces directory page has been modified.

## TEST PLAN

**Part 1: Unspecified zip folder**
1. Download this folder for testing https://hifi-content.s3.amazonaws.com/elisalj/blocks/testing/_test_upload.zip (or you can use your own zip folder)
2. Drag __test\_upload.zip_ into Interface
3. An NES model should instantiate in world
4. The info dialogue for adding files to world should disappear after a few seconds
5. Check if the _test\_upload_ directory in the Asset Browser contains four files
6. Check if the only new entity in world is the NES and not the other files
7. Check if your Local/Temp folder has any folders named "Interface-[gibberish]" (if it doesn't, that means the temporary directories cleaned themselves up properly)

**Part 2: Zip folder from Blocks**
1. Download a model from Blocks https://vr.google.com/objects
2. Drag the zip folder into world
3. The uploading behavior should be very similar to above
4. Check for your object's directory in the Asset Browser
5. Check for your entity in world (should be just the .obj in the entity list, no .mtl; both the .mtl and .obj should be in the ATP list)
6. Check your temp folder to make sure nothing got left in there

**Part 3: Does Clara still work?**
1. Open Marketplace
2. Click "See All Markets"
3. Click "Explore" under Clara
4. Pick a model you want to download
5. Click "Download to High Fidelity"
6. The model should instantiate in front of you

**Part 4: Back button in Marketplace**
1. Click "Back" until you get to the Marketplaces directory
2. Just click all the buttons to navigate around and you should be able to get back each time

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2230/HighFidelity-Beta-PR11113-5e3e87dc875932f3d46619904650559af58bf4fb.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2232/HighFidelity-Beta-PR11113-88ec7de8e0b22f6fdcdbb1c23e0f277953d81fbd.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2232/HighFidelity-Beta-PR11113-88ec7de8e0b22f6fdcdbb1c23e0f277953d81fbd.exe

File: libraries/script-engine/src/FileScriptingInterface.cpp [hunk 1]
```diff
@@ -32,26 +32,56 @@ FileScriptingInterface::FileScriptingInterface(QObject* parent) : QObject(parent
...
+void FileScriptingInterface::runUnzip(QString path, QUrl url, bool autoAdd, bool isZip) {
     qCDebug(scriptengine) << "Url that was downloaded: " + url.toString();
     qCDebug(scriptengine) << "Path where download is saved: " + path;
     QString fileName = "/" + path.section("/", -1);
     QString tempDir = path;
-    tempDir.remove(fileName);
+    if (!isZip) {
+        tempDir.remove(fileName);
+    } else {
+        QTemporaryDir blocks;
```
Comment (misslivirose) [inline libraries/script-engine/src/FileScriptingInterface.cpp]: Can we make this variable name more descriptive and not specifically related to blocks?

File: libraries/script-engine/src/FileScriptingInterface.cpp [hunk 2]
```diff
@@ -32,26 +32,56 @@ FileScriptingInterface::FileScriptingInterface(QObject* parent) : QObject(parent
...
-    QString filename = QUrl::fromLocalFile(file).toString();
-    if (file != "") {
+    QStringList fileList = unzipFile(path, tempDir);
+    QString filename = QUrl::fromLocalFile(fileList.first()).toString();
+    
+    if (filename != "") {
         qCDebug(scriptengine) << "File to upload: " + filename;
-    } else {
+    }
+    else {
```
Comment (misslivirose) [inline libraries/script-engine/src/FileScriptingInterface.cpp]: Should be on the same line as the closing bracket above it

File: libraries/script-engine/src/FileScriptingInterface.cpp [hunk 3]
```diff
@@ -32,26 +32,56 @@ FileScriptingInterface::FileScriptingInterface(QObject* parent) : QObject(parent
...
+    qCDebug(scriptengine) << "Directory to unzip: " << dirName;
+    QString target = tempDir + "/model_repo";
+    QStringList list = JlCompress::extractDir(dirName, target);
+
+    qCDebug(scriptengine) << list;
+
+    if (!list.isEmpty()) {
+        return list;
+    }
+    else {
```
Comment (misslivirose) [inline libraries/script-engine/src/FileScriptingInterface.cpp]: Should be on the same line as the closing bracket above it

File: scripts/system/html/marketplaces.html [hunk 4]
```diff
@@ -25,25 +25,25 @@ <h2 class="marketplaces-title">Marketplaces</h2>
...
-                        <p class="marketplace-tile-description">Clara.io has thousands of models available for importing into High Fidelity.  Follow these steps for the best experience:</p>
-                        <ol class="marketplace-clara-steps">
-                            <li><a id="claraSignUp" href="http://www.clara.io/signup">Create an account here </a>or log in as an existing user.</li>
-                            <li>Choose a model from the list and click &ldquo;Download to High Fidelity&rdquo;.</li>
-                        </ol>
-                        <div class="exploreButton-holder">
-                            <input class="blue exploreButton" type="button" value="Explore" id="exploreClaraMarketplace" />
-                        </div>
-                    </div>
+                <div class="exploreButton-holder">
```
Comment (misslivirose) [inline scripts/system/html/marketplaces.html]: Was moving the explore button out of the marketplace-second column here intentional?

File: libraries/script-engine/src/FileScriptingInterface.cpp (hunk 1 above)
Comment (elisa-lj11) [inline libraries/script-engine/src/FileScriptingInterface.cpp]: Changed to "zipTemp"

File: libraries/script-engine/src/FileScriptingInterface.cpp (hunk 2 above)
Comment (elisa-lj11) [inline libraries/script-engine/src/FileScriptingInterface.cpp]: not sure why my VS is automatically doing that, but changing it now

File: scripts/system/html/marketplaces.html (hunk 4 above)
Comment (elisa-lj11) [inline scripts/system/html/marketplaces.html]: image
Was intentional, the top button is when the explore button is still in the second column and the bottom button is when it is not

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2234/HighFidelity-Beta-PR11113-4701e1fce39d5113d24f331d60a37ef70b1979e6.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2234/HighFidelity-Beta-PR11113-4701e1fce39d5113d24f331d60a37ef70b1979e6.exe

**Comment by Nex-Pro** [comment]
Checking this. Win10

**Comment by Nex-Pro** [comment]
## TEST PLAN

## FAIL

### Part 1: Unspecified zip folder

Download this folder for testing https://hifi-content.s3.amazonaws.com/elisalj/blocks/testing/_test_upload.zip (or you can use your own zip folder)
Drag _test_upload.zip into Interface
An NES model should instantiate in world **SUCCESS**
The info dialogue for adding files to world should disappear after a few seconds **SUCCESS**
Check if the test_upload directory in the Asset Browser contains four files **SUCCESS**
Check if the only new entity in world is the NES and not the other files **SUCCESS**
Check if your Local/Temp folder has any folders named "Interface-[gibberish]" (if it doesn't, that means the temporary directories cleaned themselves up properly) **SUCCESS**

### Part 2: Zip folder from Blocks

Download a model from Blocks https://vr.google.com/objects
Drag the zip folder into world
The uploading behavior should be very similar to above **SUCCESS**
Check for your object's directory in the Asset Browser **SUCCESS**
Check for your entity in world (should be just the .obj, no .mtl) **FAIL I get the .obj and materials.mtl**
Check your temp folder to make sure nothing got left in there **SUCCESS**

### Part 3: Does Clara still work?

Open Marketplace
Click "See All Markets"
Click "Explore" under Clara
Pick a model you want to download
Click "Download to High Fidelity"
The model should instantiate in front of you **SUCCESS**

### Part 4: Back button in Marketplace

Click "Back" until you get to the Marketplaces directory **Initally FAIL** got stuck on the "The hunted" model page.
**Eventually, after many clicks ( more than 20 ) returned to the Marketplaces Directory**
Did not repro second time.
I am going to put that down to Having clicked to register then  cancelling registering then signing-in as had aleady registered.
Was fine on all subsequent attempts ( including after restarted interface )
Just click all the buttons to navigate around and you should be able to get back each time *SUCCESS**

**Comment by anshuman64** [comment]
@elisa-lj11 looks like there are a couple failures in the test plan

**Comment by Nex-Pro** [comment]
@anshuman64 @elisa-lj11 I'm wondering if the .mtl issue might be specific to the particular model. ( haven't read the code)

**Comment by elisa-lj11** [comment]
@Nex-Pro which model did you use?

**Comment by cain-kilgore** [comment]
Testing.

**Comment by Nex-Pro** [comment]
@cain no need to retest

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2277/HighFidelity-Beta-PR11113-a246c592775f16f554c0b7cfb048f1260349d20a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2277/HighFidelity-Beta-PR11113-a246c592775f16f554c0b7cfb048f1260349d20a.exe

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2319/HighFidelity-Beta-PR11113-a246c592775f16f554c0b7cfb048f1260349d20a.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2319/HighFidelity-Beta-PR11113-a246c592775f16f554c0b7cfb048f1260349d20a.dmg

**Pull request merged**